apps/console/src/features/identity-providers/components/settings/attribute-management/attribute-mapping-list-item.tsx

```diff
@@ -0,0 +1,301 @@
+ * are handled by onSubmit so, it is mandatory to have a submission
+ * button. If you pass editingMode={true} it will render plus icon button
+ * inline and hide button with text "Add Mapping Button" below it
+ * and input labels.
+ *
+ * @param props {AttributeMappingListIt emProps}
+ * @constructor
+ */
+export const AttributeMappingListItem: FunctionComponent<AttributeMappingListItemProps> = (
+    props: AttributeMappingListItemProps
+): ReactElement => {
+
+    const {
+        onSubmit,
+        availableAttributeList,
+        mapping,
+        editingMode
+    } = props;
+
+    const [ copyOfAttrs, setCopyOfAttrs ] = useState<Array<IdentityProviderClaimInterface>>([]);
+
+    useEffect(() => {
+        if (availableAttributeList) {
+            const copy = [ ...availableAttributeList ];
+            // When you enter into editing mode the available attribute list
+            // will not contain the mapping itself. We need to manually append
+            // it to the attrs to make it work.
+            if (editingMode && mapping && mapping.mappedValue && mapping.claim) {
+                copy.push(mapping.claim);
+            }
+            setCopyOfAttrs(copy);
+        }
+    }, [ availableAttributeList ]);
+
+    const _getListOfAvailableAttributes = () => {
+        return copyOfAttrs.map((claim, index) => ({
+            content: (
+                <Header as="h6" key={ `attribute-option-${ index }` }>
+                    <Header.Content>
+                        { claim?.displayName }
+                        <Header.Subheader>
+                            <code className="inline-code compact transparent">
+                                { claim.uri }
+                            </code>
+                        </Header.Subheader>
+                    </Header.Content>
+                </Header>
+            ),
+            key: claim?.id,
+            text: claim?.displayName,
+            value: claim?.id
+        }));
+    };
+
+    // /**
+    // * Submits the form and clears all the field values and
+    // * field states.
+    // *
+    // * @param values {object}
+    // * @param formApi {any}
+    // */
+    // const onFormSubmission = (values, formApi) => {
+    //     const newMapping = {
+    //         claim: availableAttributeList.find(
+    //             (claim) => claim.id === localClaimId
+    //         ),
+    //         mappedValue
+    //     } as IdentityProviderCommonClaimMappingInterface;
+    //     onSubmit(newMapping);
+    //     // Resets the form field values and its fields states. For some reason,
+    //     // reset() method keeps failing. It has to do something with react-final-form.
+    //     formApi.change("mappedValue", "");
+    //     formApi.resetFieldState("mappedValue");
+    //     formApi.change("localClaimId", "");
+    //     formApi.resetFieldState("mappedValue");
+    //     formApi.resetFieldState("localClaimId");
+    // };
```

Comment:
Please review and remove commented code chunks if not needed anymore.

Reply:
Removed comment in b49a285